inc/class-sitemaps-registry.php

```diff
@@ -80,7 +82,11 @@ public function remove_sitemap( $name ) {
 	 * @return array List of sitemaps.
 	 */
 	public function get_sitemaps() {
-		return $this->sitemaps;
+		$total_sitemaps = count( $this->sitemaps );
+
+		if ( $total_sitemaps <= CORE_SITEMAPS_INDEX_MAX ) {
+			return $this->sitemaps;
+		}
```

Comment:
Have you thought about what happens when more than 50k urls are in the sitemap? For example we could use `array_slice()` to return the first 50k. We could actually always call it and simplify the logic here, however I'm not sure how performant that is. https://www.php.net/manual/en/function.array-slice.php experiment with the preserve_keys parameter though.

Reply:
No, I'll confess I didn't think about that. From what I've seen from other examples, they seem to return false if there are more entries than allowed. This seems a better solution though.